### PR TITLE
Enreach some endpoints with the scope and name

### PR DESCRIPTION
**Pull request by sskserk**
Hi Patroni Guys,

here is the PR for an enhancement mentioned in the thread https://app.slack.com/client/T0FS7GCKS/C9XPYG92A/thread/C9XPYG92A-1687875160.874589

The feature significantly simplifies automatic discovery of the patroni nodes. 

We have developed a product that discovers/discovers/configures the Patroni clusters. Also, it deeply monitors insights of the posgres and the patroni itself. Hopefully together we can  make the Postgres community bigger.

### CODE REVIEW

File: patroni/api.py [hunk 1]
```diff
@@ -1263,6 +1263,7 @@ def get_postgresql_status(self, retry: bool = False) -> Dict[str, Any]:
             row = self.query(stmt.format(postgresql.wal_name, postgresql.lsn_name,
                                          postgresql.wal_flush), retry=retry)[0]
             result = {
+                'name': postgresql.name,
```
Comment (CyberDem0n) [inline patroni/api.py:1266]: If we go this way maybe we should also add the `scope`?

**Review by CyberDem0n — COMMENTED** [review]
I think it would also be worth adding the name as a tag to the `GET /metrics` endpoint: 
https://github.com/zalando/patroni/blob/89d794facc3e1fa728ccfbe6ac0323d668c8e021/patroni/api.py#L529

File: docs/rest_api.rst [hunk 2]
```diff
@@ -482,6 +486,7 @@ Let's check that the node processed this configuration. First of all it should s
 
 	$ curl -s http://localhost:8008/patroni | jq .
 	{
+          "name": "patroni1",
 	  "pending_restart": true,
```
Comment (hughcapet) [inline docs/rest_api.rst:490]:
```suggestion
      "name": "patroni1",
	  "pending_restart": true,
```

Comment (hughcapet) [inline docs/rest_api.rst:490]:
```suggestion
       "name": "patroni1",
	  "pending_restart": true,
```

File: patroni/api.py (hunk 1 above)
Comment (sskserk) [reply patroni/api.py:1266]: I'm not sure what you are mentioning. I'm chasing the "name" in the output of the "/patroni". Let's do it with a separate PR

**Review by sskserk — COMMENTED** [review]
QnA

File: patroni/api.py (hunk 1 above)
Comment (CyberDem0n) [inline patroni/api.py:1266]: It doesn't deserve a separate PR.
```suggestion
                'scope': postgresql.scope,
                'name': postgresql.name,
```

**Review by sskserk — COMMENTED** [review]
Addressed comment

**Comment by CyberDem0n** [comment]
@sskserk what about https://github.com/zalando/patroni/pull/2846#pullrequestreview-1608775982?
This is a standard prometheus endpoint...

**Comment by sskserk** [comment]
@CyberDem0n  prometheus endpoint is enhanced

**Comment by hughcapet** [comment]
Now `/metrics` endpoint docs also need to be extended. Also please check pyright errors

**Comment by CyberDem0n** [comment]
@sskserk this is not what I meant.
We need to add `name` to `labels`:
```diff
-        scope_label = '{{scope="{0}"}}'.format(patroni.postgresql.scope)
+        labels = f'{{scope="{patroni.postgresql.scope}",name="{patroni.postgresql.name}"}}'
```

And of course rename all occurrences of `scope_label` to `labels` down the method.

**Comment by CyberDem0n** [comment]
A few more changes are needed:
1. Add scope to all examples of monitoring endpoints
2. update labels (add name) in the example of Prometheus output: https://github.com/zalando/patroni/blob/eb6bbea33853d241509ec67216296e65d1303f7f/docs/rest_api.rst?plain=1#L286-L349

And when looking at the [documentation](https://github.com/zalando/patroni/blob/eb6bbea33853d241509ec67216296e65d1303f7f/docs/rest_api.rst?plain=1#L351-L360?) I also wonder if we should add scope to the `GET /cluster` endpoint [here](https://github.com/zalando/patroni/blob/eb6bbea33853d241509ec67216296e65d1303f7f/patroni/utils.py#L802)?

File: docs/rest_api.rst [hunk 3]
```diff
@@ -279,70 +283,70 @@ Retrieve the Patroni metrics in Prometheus format through the ``GET /metrics`` e
 	
 	# HELP patroni_version Patroni semver without periods. \
 	# TYPE patroni_version gauge
-	patroni_version{scope="batman"} 020103
+	patroni_version{scope="batman", name="patroni1"} 020103
```
Comment (CyberDem0n) [inline docs/rest_api.rst:290]: no space character needed
```suggestion
	patroni_version{scope="batman",name="patroni1"} 020103
```
please remove it from the remaining lines.

**Review by CyberDem0n — COMMENTED** [review]
1. you already added "name" to monitoring endpoints examples: https://github.com/zalando/patroni/pull/2846/files#diff-c10e80ec2654546cb5d06ed2aee8a7de61371f731c065d2afc623c7c13728291R101. Similarly you need to add "scope"
2. add "scope" here: https://github.com/zalando/patroni/blob/eb6bbea33853d241509ec67216296e65d1303f7f/patroni/utils.py#L802 and update respective [documentation](https://github.com/zalando/patroni/blob/eb6bbea33853d241509ec67216296e65d1303f7f/docs/rest_api.rst?plain=1#L351-L360).

**Comment by sskserk** [comment]
I'm not sure how to obtain the 'scope' in the function which you mention in relation to the code ' ret: Dict[str, Any] = {'members': []} '

File: patroni/utils.py [hunk 4]
```diff
@@ -809,7 +809,7 @@ def cluster_as_json(cluster: 'Cluster', global_config: Optional['GlobalConfig']
             role = 'replica'
 
         state = (m.data.get('replication_state', '') if role != 'leader' else '') or m.data.get('state', '')
-        member = {'name': m.name, 'role': role, 'state': state, 'api_url': m.api_url}
+        member = {'scope', global_config.scope, 'name': m.name, 'role': role, 'state': state, 'api_url': m.api_url}
```
Comment (CyberDem0n) [inline patroni/utils.py:812]: Oh, you are right. Here the scope isn't available :(
We need to do it here: https://github.com/zalando/patroni/blob/6b7f914da7fd4974ca091c8f55c769b2ad0f58a3/patroni/api.py#L452

Comment (sskserk) [reply patroni/utils.py]: Not sure what you are mentioning. This function just sends the response.

Comment (sskserk) [reply patroni/utils.py]: Please suggest the change. It will take significantly less time and efforts.

Comment (CyberDem0n) [reply patroni/utils.py]: Yes, but we can always modify it before sending:
```python
response = cluster_as_json(cluster, global_config)
response['scope'] = self.server.patroni.postgresql.scope
self._write_json_response(200, response) 
```

File: docs/rest_api.rst [hunk 5]
```diff
@@ -356,6 +364,7 @@ Cluster status endpoints
     {
       "members": [
         {
+          "scope": "demo",
```
Comment (CyberDem0n) [inline docs/rest_api.rst:367]: scope should be on the cluster level

**Comment by CyberDem0n** [comment]
@sskserk sometimes they fail, MacOS VMs could be very slow :(

**Comment by sskserk** [comment]
What are the next actions? @CyberDem0n

**Comment by sskserk** [comment]
All tests passed

**Comment by CyberDem0n** [comment]
Oh, @sskserk I am very sorry, but we'll have to redo some of the things :(
I totally missed the fact that the `scope` is already available in the response since many years: https://github.com/zalando/patroni/commit/704b29e6868eff28ffdea12d044e962b90d177e3 in the `{"patroni":{"scope":"$SCOPE"}}` and I falsely insisted on adding yet another one :(
Thanks @hughcapet for pointing it out. As a result I start thinking whether the "name" should also be put under "patroni"?
Any thoughts about it?

**Comment by sskserk** [comment]
@CyberDem0n,
Do you want to move the "name" into the "patroni" struct? No problem at all. I'm chasing for the "name". It doesn't make a big difference where it will be available

**Review by sskserk — COMMENTED** [review]
format

**Comment by sskserk** [comment]
These is something wrong with the runner

File: patroni/api.py [hunk 6]
```diff
@@ -1262,7 +1269,7 @@ def get_postgresql_status(self, retry: bool = False) -> Dict[str, Any]:
 
             row = self.query(stmt.format(postgresql.wal_name, postgresql.lsn_name,
                                          postgresql.wal_flush), retry=retry)[0]
-            result = {
+            result =
```
Comment (CyberDem0n) [inline patroni/api.py:1272]:
```suggestion
            result = {
```

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Comment by CyberDem0n** [comment]
Merged, thank you @sskserk!

**Comment by sskserk** [comment]
@CyberDem0n,
 thanks for merging the PR.
When will it be incorporated into a release?

We will start  recommending it to our customers.

**Comment by sskserk** [comment]
Hi @CyberDem0n,

When do you plan to merge the feature into a release? We are are just looking forward to installing it into production.

**Comment by adolsalamanca** [comment]
hello
Was this PR skipped on purpose on release v3.1.2?
We're interested on this.

Thanks!

**Comment by adolsalamanca** [comment]
hello
Was this PR skipped on purpose on release `v3.1.2`?
We're interested on this.

Thanks!

**Comment by hughcapet** [comment]
This feature will be included into 3.2, 3.1.2 is a bugfix only release
